ffi/derive/src/convert.rs

```diff
@@ -0,0 +1,315 @@
+            type Store = ();
+
+            unsafe fn try_from_repr_c(source: <Self as iroha_ffi::TryFromReprC<'itm>>::Source, _: &mut <Self as iroha_ffi::TryFromReprC<'itm>>::Store) -> iroha_ffi::Result<Self> {
+                source.as_ref().ok_or(iroha_ffi::FfiReturn::ArgIsNull)
+            }
+        }
+        impl<'itm> iroha_ffi::TryFromReprC<'itm> for &'itm mut #name {
+            type Source = *mut #name;
+            type Store = ();
+
+            unsafe fn try_from_repr_c(source: <Self as iroha_ffi::TryFromReprC<'itm>>::Source, _: &mut <Self as iroha_ffi::TryFromReprC<'itm>>::Store) -> iroha_ffi::Result<Self> {
+                source.as_mut().ok_or(iroha_ffi::FfiReturn::ArgIsNull)
+            }
+        }
+
+        impl<'slice> iroha_ffi::slice::TryFromReprCSliceRef<'slice> for #name {
+            type Source = iroha_ffi::slice::SliceRef<'slice, <&'slice Self as iroha_ffi::TryFromReprC<'slice>>::Source>;
+            type Store = Vec<Self>;
+
+            unsafe fn try_from_repr_c(source: <Self as iroha_ffi::slice::TryFromReprCSliceRef<'slice>>::Source, store: &'slice mut <Self as iroha_ffi::slice::TryFromReprCSliceRef<'slice>>::Store) -> iroha_ffi::Result<&'slice [Self]> {
+                let source = source.into_rust().ok_or(iroha_ffi::FfiReturn::ArgIsNull)?;
+
+                for elem in source {
+                    store.push(Clone::clone(iroha_ffi::TryFromReprC::try_from_repr_c(*elem, &mut ())?));
+                }
+
+                Ok(store)
+            }
+        }
+    }
+}
+
+fn derive_try_from_ffi_for_item(_: &Ident) -> TokenStream2 {
+    quote! {
+        // TODO:
+    }
+}
+
+fn derive_try_from_ffi_for_fieldless_enum(
+    enum_name: &Ident,
+    enum_: &syn::DataEnum,
+    repr: &[syn::NestedMeta],
+) -> TokenStream2 {
+    let variant_names: Vec<_> = enum_.variants.iter().map(|v| &v.ident).collect();
+    let discriminant_values = variant_discriminants(enum_);
+
+    let ffi_type = enum_size(enum_name, repr);
+    let (discriminants, discriminant_names) =
+        variant_names.iter().zip(discriminant_values.iter()).fold(
+            <(Vec<_>, Vec<_>)>::default(),
+            |mut acc, (variant_name, discriminant_value)| {
+                let discriminant_name = Ident::new(
+                    &format!("{}__{}", enum_name, variant_name).to_uppercase(),
+                    proc_macro2::Span::call_site(),
+                );
+
+                acc.0.push(quote! {
+                    const #discriminant_name: #ffi_type = #discriminant_value;
+                });
+                acc.1.push(discriminant_name);
+
+                acc
+            },
+        );
+
+    quote! {
+        impl<'itm> iroha_ffi::TryFromReprC<'itm> for #enum_name {
+            type Source = #ffi_type;
+            type Store = ();
+
+            unsafe fn try_from_repr_c(source: <Self as iroha_ffi::TryFromReprC<'itm>>::Source, _: &mut <Self as iroha_ffi::TryFromReprC<'itm>>::Store) -> iroha_ffi::Result<Self> {
```

Comment:
```suggestion
            unsafe fn try_from_repr_c(
                source: <Self as iroha_ffi::TryFromReprC<'itm>>::Source,
                _: &mut <Self as iroha_ffi::TryFromReprC<'itm>>::Store
            ) -> iroha_ffi::Result<Self> {
```

Reply:
Same for other lines like that

Reply:
Sad, that `cargo fmt` does not format macros although, I believe, there are reasons for that

Reply:
Rust has insignificant whitespace. Proc. macros in particular could have significant whitespace. So altering code might cause a problem. You can format region in most text editors though, so it should be done selectively.